### PR TITLE
Cmake and build fixes.

### PR DESCRIPTION
**Pull request by marcinslusarz**
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemkv/44)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -4,18 +4,51 @@ project(pmemkv)
...
+if(PKG_CONFIG_FOUND)
+	pkg_check_modules(PMEMOBJ++ REQUIRED libpmemobj++)
+else()
+	find_package(PMEMOBJ++ REQUIRED)
+endif()
+
+include_directories(${PMEMOBJ++_INCLUDE_DIRS})
+link_directories(${PMEMOBJ++_LIBRARY_DIRS})
 
 add_library(pmemkv ${SOURCE_FILES})
```
Comment (GBuella) [inline CMakeLists.txt]: Also, this should be:
```add_library(pmemkv SHARED ${SOURCE_FILES})```

Comment (GBuella) [inline CMakeLists.txt]: Although, it the build seems to work like this as well. It only creates a ```libpmemkv.a``` file - I suspect a shared object is the intended target, right?

Comment (tomaszkapela) [inline CMakeLists.txt]: Taken from the docs:
> If no type is given explicitly the type is STATIC or SHARED based on whether the current value of the variable BUILD_SHARED_LIBS is ON.

So I think it's fine. Let the user decide if it's not a problem.

Comment (tomaszkapela) [inline CMakeLists.txt]: Even better, we could do:
`option(BUILD_SHARED_LIBS "Build shared libraries" ON)`

Comment (GBuella) [inline CMakeLists.txt]: Oh, yes, that makes a lot of sense now.

Comment (RobDickinson) [inline CMakeLists.txt]: Thank you so much for all the help with the cmake build -- this is definitely not one of my strengths.

**Pull request merged**